src/Avalonia.Controls/Shapes/Arc.cs

```diff
@@ -0,0 +1,148 @@
+                {
+                    _arcSegment
+                },
+                IsClosed = false
+            };
+
+            _arcGeometry = new PathGeometry()
+            {
+                Figures = 
+                {
+                    _arcFigure
+                }
+            };
+        }
+
+        public double StartAngle
+        {
+            get { return GetValue(StartAngleProperty); }
+            set { SetValue(StartAngleProperty, value); }
+        }
+
+        public double EndAngle
+        {
+            get { return GetValue(EndAngleProperty); }
+            set { SetValue(EndAngleProperty, value); }
+        }
+        
+        protected override Geometry CreateDefiningGeometry()
+        {
+            double angle1 = DegreesToRad(StartAngle);
+            double angle2 = DegreesToRad(EndAngle);
+            
+            double startAngle = Math.Min(angle1, angle2);
+            double endAngle = Math.Max(angle1, angle2);
+
+            double normStart = RadToNormRad(startAngle);
+            double normEnd = RadToNormRad(endAngle);
+
+            var rect = new Rect(Bounds.Size);
+
+            if ((normStart == normEnd) && (startAngle != endAngle)) //complete ring
+            {
+                return new EllipseGeometry(rect.Deflate(StrokeThickness / 2));
+            }
+            else if ((normStart == normEnd) && (startAngle == endAngle)) //empty
+            {
+                return _emptyGeometry;
+            }
+            else //partial ring
+            {
+                var deflatedRect = rect.Deflate(StrokeThickness / 2);
+
+                double centerX = rect.Center.X;
+                double centerY = rect.Center.Y;
+
+                double radiusX = deflatedRect.Width / 2;
+                double radiusY = deflatedRect.Height / 2;
+                
+                double angleGap = RadToNormRad(endAngle - startAngle);
+
+                Point startPoint = GetRingPoint(radiusX, radiusY, centerX, centerY, startAngle);
+                Point endPoint = GetRingPoint(radiusX, radiusY, centerX, centerY, endAngle);
+                
+                _arcFigure.StartPoint = startPoint;
+                
+                _arcSegment.Point = endPoint;
+                _arcSegment.IsLargeArc = angleGap >= Math.PI;
+                _arcSegment.Size = new Size(radiusX, radiusY);
+                
+                return _arcGeometry;
+            }
+        }
+
+        public override void Render(DrawingContext ctx)
+        {
+            double angle1 = DegreesToRad(StartAngle);
+            double angle2 = DegreesToRad(EndAngle);
+            
+            double startAngle = Math.Min(angle1, angle2);
+            double endAngle = Math.Max(angle1, angle2);
+
+            double normStart = RadToNormRad(startAngle);
+            double normEnd = RadToNormRad(endAngle);
+
+
+            if ((normStart == normEnd) && (startAngle == endAngle)) //empty
+            {
+
```

Comment:
I always like to leave the comment for why this case is empty and doesn't need to be handled inside the braces. Just helps for readability and no one wonders why there is a set of empty braces.

Reply:
Woah...that's actually not what I meant by "empty" at all - I was referring to the arc itself being "empty" (e.g. both angles are the same)...the term was carried over from the `ProgressRing`,  where it actually makes sense to describe it as "empty", "full", "partially-filled", etc.. Weird coincidence, huh?

You make a good point though. I'll fix that comment and elaborate as you described.

Reply:
Yes, from the code I knew you meant the arc was empty. Just my poor phrasing above causing confusion between empty braces and empty arc.